dissect/cstruct/types/pointer.py

```diff
@@ -16,7 +16,7 @@ class Pointer(int, BaseType):
     _context: dict[str, Any]
     _value: BaseType
 
-    def __new__(cls, value: int, stream: BinaryIO, context: dict[str, Any] = None) -> Pointer:
+    def __new__(cls, value: int, stream: BinaryIO | None, context: dict[str, Any] = None) -> Pointer:
         obj = super().__new__(cls, value)
         obj._stream = stream
```

Comment:
If stream can be `None`, then the `_stream` field should be `BinaryIO | None` I think